modules/lfs/http_client.go

```diff
@@ -134,71 +137,84 @@ func (c *HTTPClient) performOperation(ctx context.Context, objects []Pointer, dc
 		return fmt.Errorf("TransferAdapter not found: %s", result.Transfer)
 	}
 
+	errGroup, groupCtx := errgroup.WithContext(context.Background())
 	for _, object := range result.Objects {
-		if object.Error != nil {
-			log.Trace("Error on object %v: %v", object.Pointer, object.Error)
-			if uc != nil {
-				if _, err := uc(object.Pointer, object.Error); err != nil {
-					return err
-				}
-			} else {
-				if err := dc(object.Pointer, nil, object.Error); err != nil {
-					return err
-				}
-			}
-			continue
-		}
+		func(groupCtx context.Context, object *ObjectResponse, dc DownloadCallback, uc UploadCallback, transferAdapter TransferAdapter) {
```

Comment:
https://go.dev/blog/loopvar-preview

No need such trick?

Reply:
Please advise, are you suggesting that I can just do this?:
```
for _, object := range result.Objects {
  go performSingleOperation(groupCtx, object, dc, uc, transferAdapter)
}
```

Reply:
Yup, I think so if it is right.

Reply:
> Admittedly I used this pattern because I'd seen it in practice elsewhere,

In history, it is due to https://go.dev/blog/loopvar-preview https://github.com/go-gitea/gitea/pull/32369/files#r1821738812

> not because I fully understood _why_ it was in use

Well, that's why I would like to clarify the problem. I don't philosophy is that "do not write the code which you do not understand"



> but `errGroup.Go` only takes a function without arguments; you cannot pass arguments to functions inside errGroup.

https://github.com/go-gitea/gitea/pull/32369/files#r1823705885

I do not see why it is a problem.

```
for _, object := range result.Objects {
	errGroup.Go(func() error {
		return performSingleOperation(groupCtx, object, dc, uc, transferAdapter)
	})
}
```